pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/IngestionUtils.java

```diff
@@ -110,8 +112,6 @@ public static SegmentGeneratorConfig generateSegmentGeneratorConfig(TableConfig
 
     // apply config override provided by user.
     Map<String, String> batchConfigMap = new HashMap<>(batchIngestionConfig.getBatchConfigMaps().get(0));
```

Comment:
Q: is this defensive copy needed? I found the old code didn't do the copy.

Reply:
good catch this is not needed.